### ISSUE TITLE
Set Suggest Proof Using does not give sufficiently descriptive messages

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#7877]
#### Version

8.8.0

#### Description of the problem

Sometime in the not-too-distant past, `Set Suggest Proof Using` stopped listing module names.  This makes it nearly impossible to automatically script replacing a file with the suggested `Proof using` directive.  Furthermore, the `.aux` file gives the location of `Qed`, and it is non-trivial to locate the corresponding start-of-theorem before that.  It'd be quite useful to have an annotation in the `.aux` file that gives the location of the `.` at the end of the `Theorem`/`Definition`/`Next Obligation.`/... vernacular.  Alternatively, please ship a tool with Coq or a `coq_makefile`-made-`Makefile` target that implements the `Set Suggest Proof Using` suggestions.

cc @gares

**Comment by ejgallego** [comment]
> I guess one could restore the printing of module names, I actually don't know why it was removed.

Previously this feature was using a debug printer so it printed raw terms, now it is using the normal printer. Anyways what Jason suggests is not robust so please let's not go this route.

If you want to talk with Coq from a script please use a tool that supports that.

**Comment by gares** [comment]
> ltac:(Goal True). ltac:(Qed).

At the risk of looking dumb, is this valid Coq?

**Comment by SkySkimmer** [comment]
Sure 
~~~coq

Ltac Goal _ := idtac.
Ltac Qed := idtac.

Goal True.
  ltac:(Goal True). ltac:(Qed). 
~~~

**Comment by JasonGross** [comment]
I did, in fact, run the code I used here through Coq 8.8 before commenting.  I wouldn't trust myself to get the corner cases right otherwise.  (e.g., I believe `Definition Goal := True` passes but `Definition Theorem := True` fails.)

**Comment by gares** [comment]
OK for the trivia, I think I will collect some ideas I have on the .aux topic and mention this issue there

**Comment by gares** [comment]
If the print is now in Lemmas I think I can print the full kernel name of the object using a high level pretty printer.

**Comment by SkySkimmer** [comment]
Strictly speaking the print is in vernac/proof_using, but since it's only called from lemmas that's the only upper constraint on its position.

**Comment by silene** [comment]
@gares Do you intend to work on that for 8.9? Or should the milestone be moved?

**Comment by gares** [comment]
Move the milestone plz

**Comment by ejgallego** [comment]
I tried to fix this but I couldn't figure out what the proper printers are.

**Comment by SkySkimmer** [comment]
>I tried to fix this but I couldn't figure out what the proper printers are.

Note the the issue title seems to be just about proof using but then the issue OP says
>It'd be quite useful to have an annotation in the .aux file that gives the location of the . at the end of the Theorem/Definition/Next Obligation./... vernacular.

**Comment by ejgallego** [comment]
I think that if you want to do programmatic manipulation of v files is better to use tools that support that use case such as serapi instead of relying on text hacks, but I dunno.

**Comment by gares** [comment]
I agree, the message in question is for user consumption. For that use case, I believe it is quiet OK.

IIRC the .aux file contains a more machine friendly version of the suggestion, that one could be used for tools.
